### PR TITLE
Development => Master branch

### PR DESCRIPTION
**Pull request by Grant-Steinfeld**
@MaxShapiro32 please review our changes just discussed and if all looks good please merge up to master branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,372 @@
...
+
+* [IBM Cloud RedHat Openshift version 4](https://www.ibm.com/cloud/openshift): Red Hat® OpenShift® on IBM Cloud™ is a fully managed OpenShift service that leverages the enterprise scale and security of the IBM Cloud.
+* [Swagger](https://swagger.io/): A framework of API developer tools for the OpenAPI Specification that enables development across the entire API lifecycle.
+
+
+
+## Featured technologies
+
+* [Container Orchestration](https://www.ibm.com/cloud-computing/bluemix/containers): Automating the deployment, scaling and management of containerized applications.
+* [Microservices](https://www.ibm.com/developerworks/community/blogs/5things/entry/5_things_to_know_about_microservices?lang=en): Collections of fine-grained, loosely coupled services using a lightweight protocol to provide building blocks in modern application composition in the cloud.
```
Comment (MaxShapiro32) [inline README.md]: Update the links since they reroute for the components/technologies

File: README.md [hunk 2]
```diff
@@ -1,2 +1,372 @@
...
+* [IBM Cloud Account](https://cloud.ibm.com/registration)
+* [IBM RedHat OpenShift 4 Cluster](https://cloud.ibm.com/kubernetes/catalog/openshiftcluster)
+* OpenShift CLI tool [oc](https://cloud.ibm.com/docs/openshift?topic=openshift-openshift-cli#cli_oc)
+
+# Steps 
+
+Follow these steps to set up and run this code pattern locally and on the cloud. The steps are described in detail below.
+
+1. [Clone the repo](#1-clone-the-repo)
+2. [Build a docker image, then run it locally](#3-Build-a-docker-image-then-run-it-locally)
```
Comment (MaxShapiro32) [inline README.md]: Fix the links for these steps

File: README.md [hunk 3]
```diff
@@ -1,2 +1,372 @@
...
+3. [Deploy to IBM RedHat OpenShift 4 Cluster](#4-deploy-to-ibm-cloud)
+
+### 1. Clone the repo
+
+Clone the `currencyexchange` repo locally. In a terminal, run:
```
Comment (MaxShapiro32) [inline README.md]: Clone the repo locally.

File: README.md [hunk 4]
```diff
@@ -1,2 +1,372 @@
...
+The UBI comes in a few flavors:
+
+1.  You can choose one of  three base images (`ubi`, `ubi-minimal` and `ubi-init`)
+1.  Or Language specific runtime images (e.g. `node.js`, `python`, etc.)
+
+UBI allows one to use associated packages via `YUM repositories` which satisfy common application dependencies, like `httpd` (apache web server) etc.
+
+
+
+### Take a look at our [Dockerfile](./Dockerfile) and notice the `FROM` directive is using the UBI version 8 ( core of RedHat 8) base image.
```
Comment (MaxShapiro32) [inline README.md]: Remove space between parenthesis

File: README.md [hunk 5]
```diff
@@ -1,2 +1,372 @@
...
+
+```yaml
+FROM registry.access.redhat.com/ubi8/ubi
+```
+
+Now let's build this docker image with the `RedHat UBI`.
+
+
+1. Make sure you are at the root of this application.
```
Comment (MaxShapiro32) [inline README.md]: Root directory

File: README.md [hunk 6]
```diff
@@ -1,2 +1,372 @@
...
+2. Provision an [IBM RedHat OpenShift 4 Service](https://cloud.ibm.com/kubernetes/catalog/openshiftcluster)
+and follow the set of instructions for creating a Container and Cluster.
+
+### There are 2 ways to deploy the image to OpenShift.
+
+1. Using the Developer Tools CLI
+2. OpenShift web console
+
+
+#### Step 1) Using the [Developer Tools CLI](https://www.ibm.com/cloud/cli):
```
Comment (MaxShapiro32) [inline README.md]: OC CLI not IBM Cloud CLI

File: README.md [hunk 7]
```diff
@@ -1,2 +1,372 @@
...
+
+Login to your OpenShift 4 cluster
+
+![2 ways to connect to OpenShift cluster](doc/images/OpenShift-connection-to-cluster-2-ways.png)
+
+click the Actions/Connect via CLI ( annotated with a number(1) above ) and follow the instructions:
+
+use `oc login ... ` to login to your cluster, for example 
+```sh
+oc login --token=X8bjO-ROAhGUx8S9pvg6767574ysuG9SSgSI6hyg --server=https://c108-e.us-northwest.containers.cloud.ibm.com:31007
```
Comment (MaxShapiro32) [inline README.md]: Does this need to be sanitized?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,372 @@
...
+```
+</details>
+
+
+2. Provision an [IBM RedHat OpenShift 4 Service](https://cloud.ibm.com/kubernetes/catalog/openshiftcluster)
+and follow the set of instructions for creating a Container and Cluster.
+
+### There are 2 ways to deploy the image to OpenShift.
+
+1. Using the Developer Tools CLI
```
Comment (MaxShapiro32) [inline README.md]: Maybe link these to where they are in the README

File: README.md [hunk 9]
```diff
@@ -1,2 +1,372 @@
...
+oc expose svc/currencyexchange-py
+
+#expected output
+# route.route.openshift.io/currencyexchange-py exposed
+
+#get the external URL to access the microservice
+oc status
+
+```
+![getting the external url](doc/images/OpenShift-get-external-url-cli.png)
```
Comment (MaxShapiro32) [inline README.md]: I can make out the sanitized part

File: README.md [hunk 10]
```diff
@@ -1,2 +1,372 @@
...
+
+So copy and paste the url indicated in yellow highlight above into your favorite web browser and voila!  You should see:
+
+![OpenShift url shows swagger and exchange rate conversion executes as expected](doc/images/OpenShift-url-navigate-to-shows-swagger-success.png)
+
+
+Looking at the OpenShift Web console we can now see our microservice all setup and running nicely.
+
+![OpenShift Web console up and running](doc/images/OpenShift-webconsole-after-CLI-image-deploy.png)
```
Comment (MaxShapiro32) [inline README.md]: Keeping this without sanitizing?

File: README.md [hunk 11]
```diff
@@ -1,2 +1,372 @@
...
+
+You should now see Python Flask swagger API interactive web page, where you can manually test the Currency Conversion as seen below. 
+
+![OpenShift url shows swagger and exchange rate conversion executes as expected](doc/images/OpenShift-url-navigate-to-shows-swagger-success.png)
+
+
+
+### Congratulations!  
+
+You have now successfully created an microservice using the `RedHat Universal Base Image (UBI)` on `OpenShift 4`
```
Comment (MaxShapiro32) [inline README.md]: You have now successfully created and deployed a microservice using the RedHat Universal Base Image (UBI) in an OpenShift 4 cluster on IBM Cloud

File: requirements.txt [hunk 12]
```diff
@@ -0,0 +1,4 @@
+flask
```
Comment (MaxShapiro32) [inline requirements.txt]: Add versions

File: src/services/README.md [hunk 13]
```diff
@@ -0,0 +1,3 @@
+# Services directory
```
Comment (MaxShapiro32) [inline src/services/README.md]: Is this file needed?

File: .travis.yml [hunk 14]
```diff
@@ -0,0 +1,9 @@
+language: python
```
Comment (MaxShapiro32) [inline .travis.yml]: Travis is failing so I can't merge anyways

Comment (Grant-Steinfeld) [reply .travis.yml]: yes it's a python thing - I got circleCI working last night on another repo - so can try fix this
